client/java/src/test/java/io/openlineage/client/transports/CompositeTransportTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.transports;
+
+import static io.openlineage.client.Events.runEvent;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import io.openlineage.client.OpenLineage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.NonNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+// Fake transport base class for testing
+abstract class FakeTransport extends Transport {
+  private boolean emitted;
+
+  public FakeTransport() {
+    super(Type.NOOP); // Using NOOP or any placeholder type
+  }
+
+  public boolean isEmitted() {
+    return emitted;
+  }
+
+  @Override
+  public void emit(@NonNull OpenLineage.RunEvent runEvent) {
+    emitted = true; // Set to true when emit is called
+  }
+
+  @Override
+  public void emit(@NonNull OpenLineage.DatasetEvent datasetEvent) {
+    emitted = true; // Set to true when emit is called
+  }
+
+  @Override
+  public void emit(@NonNull OpenLineage.JobEvent jobEvent) {
+    emitted = true; // Set to true when emit is called
+  }
+}
+
+// FakeTransportA extending the base class
+class FakeTransportA extends FakeTransport {}
+
+// FakeTransportB extending the base class
+class FakeTransportB extends FakeTransport {}
+
+// Mock transport configuration class for FakeTransportA
+class FakeTransportConfigA implements TransportConfig {}
+
+// Mock transport configuration class for FakeTransportB
+class FakeTransportConfigB implements TransportConfig {}
+
+class CompositeTransportTest {
+
+  private static FakeTransportA fakeTransportA;
+  private static FakeTransportB fakeTransportB;
+  private CompositeConfig compositeConfig;
+
+  @BeforeEach
+  void setUp() {
+    // Initialize fake transports
+    fakeTransportA = spy(new FakeTransportA());
+    fakeTransportB = spy(new FakeTransportB());
+
+    // Mock configuration for the CompositeTransport
+    Map<String, Object> fakeTransportAConfig = new HashMap<>();
+    fakeTransportAConfig.put("type", "fakeA");
+    Map<String, Object> fakeTransportBConfig = new HashMap<>();
+    fakeTransportBConfig.put("type", "fakeB");
+
+    compositeConfig =
+        new CompositeConfig(List.of(fakeTransportAConfig, fakeTransportBConfig), true);
```

Comment:
`List.of` comes from Java 9. Does it work? In `compileTestJava` (Gradle) we have source compatibility set to java 8

Reply:
Apart from these little comments/doubts, looks good to me.

Reply:
Arrays.asList()

Reply:
Changed as per @mobuchowski's suggestion.